language-design.md

```diff
@@ -38,18 +160,89 @@ How do the themes of _Growing a Language_ relate to the "sound lab" we did this
 
 **Response**
 
-
+Our main takeaway from _Growing a Language_ was that a good language should
+provide users with a small and useful set of building blocks.  From there, the
+programmer should be able to compose the available tools to suit their needs. We
+tried to use this mentality when adapting the Sound library to a more usable
+API.  First, we created a "Sound" class which contained all of the methods for
+editing sounds.  We tried to be consistent--each method took in a Sound (self)
+along with other parameters and returned a new Sound.  This way, the user could
+easily compose methods using calls like `mySound.reverse().overlay(otherSound)`.
+
+In addition, _Growing a Language_ showed the importance of keeping an API as 
+contained as possible, and only including options that would be relevant to 
+many users.  On pages 10 and 11, Steele shows many examples of data types, each
+increasingly obscure.  After describing them, Steele asks "So should we make
+'x' a type in the Java programming language?"  In the end, he concludes that
+language designers should not add these rare features because "It would not
+be fair to weigh down all programmers with the need to have or to learn all the
+words for all niche uses" (Steele, 12).  This idea ties with the theme of making
+a language/API exactly as small as it should be, and no smaller [Bloch, 2006]. 
+We tried to follow this maxim when designing our API for the sound lab. 
+All of the methods performed a single task, which allowed users to combine
+functions effectively.  We tried to keep the methods as segmented as possible,
+including options like "play", "overlay," "reverse," and "change speed".  We did
+not, for example, add a method that would both overlay two sounds and
+reverse them, as this would not have been useful for most users.
+
+Another idea that _Growing a Language_ emphasized was that users should be able
+to make their own functions that look like primitives of the language. 
+According to Steele, 
+> APL was designed by one man, a smart man—and I love APL—but it had a flaw that
+> I think has all but killed it: there was no way for a user to grow the
+> language in a smooth way (Steele, 6)
+
+In retrospect, we failed to consider this point when designing our Sound 
+API.  Our design involved creating a Sound class with two data members, the 
+sound samples and the sampling rate.  Our goal was to keep these data members 
+private and only expose users to the class's methods.  (Bloch includes this 
```

Comment:
I don't think Python supports private data members. I did some research and I found [this post](https://plus.google.com/115212051037621986145/posts/7wpbQTPRWft) by Guido van Rossum (Python's "Benevolent Dictator for Life"), which explains some of his reasoning behind not supporting private data members. In fact, Python seems to encourage directly accessing data members. Getters and setters are not considered "pythonic".

I think Bloch's maxim that data members should be private is a principle shared by many people in CS, so it's interesting how the Python community prefers otherwise. I'm biased towards Python, so I'm not sure if this is a count against Python being a good language.